core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/EnvelopeSimContext.kt

```diff
@@ -28,19 +29,42 @@ constructor(
          * *after* the end of the path (or right at the end).
          */
         val dangerPointOffsets: List<Offset<TravelledPath>>,
+        /**
+         * List of ETCS-block detector offsets for every block on the path. Starts at the start of
+         * the path, can end after the end of the path.
+         */
+        val detectorOffsets: List<Offset<TravelledPath>>
     ) {
         /**
-         * Returns the SVL location: next buffer stop or switch, whichever is closest. If there is
-         * any.
+         * Returns the next danger point location: next buffer stop or switch, whichever is closest.
+         * If there is any.
+         */
+        fun getDangerPoint(offset: Offset<TravelledPath>): Offset<TravelledPath>? {
+            return dangerPointOffsets.firstOrNull { it >= offset }
+        }
+
+        /**
+         * On a closed signal of for an interlocking signal, a danger point is always located less
```

Comment:
```suggestion
         * On a closed signal or for an interlocking signal, a danger point is always located less
```

(not 100% sure I interpreted it right, but the current phrasing is wrong either way)

Reply:
```suggestion
         * On a closed signal stop or for an interlocking signal, a danger point is always located less
```

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/12457/commits/e3ca3fe9ad1ab72dbdb4807beea8957011852d32.